packages/orchestration/src/orchestration-response.ts

```diff
@@ -51,6 +55,54 @@ export class OrchestrationResponse {
     return choice?.message?.content;
   }
 
+  /**
+   * Messages that can be used for subsequent prompts as message history.
+   * @param choiceIndex - The index of the choice to parse.
+   * @returns A list of all messages.
+   */
+  getAllMessages(choiceIndex = 0): ChatMessages {
+    const messages = (this.data.module_results.templating ?? []).map(
+      (message: ChatMessage) => {
+        const isMultiContent = Array.isArray(
+          (message as MultiChatMessage).content
+        );
+        return isMultiContent
+          ? this.handleMultiChatMessage(message as MultiChatMessage)
+          : this.handleSingleChatMessage(message as SingleChatMessage);
+      }
+    );
+
+    messages.push({
+      role: 'assistant',
+      content:
+        this.getChoices().find(c => c.index === choiceIndex)?.message.content ??
```

Comment:
[q] Ummm I am not sure if index cannot be found, we should push empty string in to messages. It is possible that the last message got filtered and in this case, we should maybe leave the message out?